test/ros_helpers_test.cpp

```diff
@@ -0,0 +1,41 @@
+#include <common_robotics_utilities/ros_helpers.hpp>
+
+#include <gtest/gtest.h>
+
+namespace common_robotics_utilities
+{
+namespace ros_helpers
+{
+namespace
+{
+
+#if COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 2
+using RosDuration = rclcpp::Duration;
+#elif COMMON_ROBOTICS_UTILITIES__SUPPORTED_ROS_VERSION == 1
+using RosDuration = ros::Duration;
+#endif
+
+GTEST_TEST(ROSHelpersTest, SetMessageTimestamps)
+{
+  VisualizationMarkerArray marker_array;
+  marker_array.markers.resize(5);
+  // Note that the ROS time type depends on which ROS variant is in use.
+  // Get marker base timestamp (ensuring clock sources match).
+  const RosTime base_stamp = marker_array.markers[0].header.stamp;
```

Comment:
I'm not understanding part of this - my understanding of ROS messages in C++ is that all members are default-constructed, so why would `const RosTime base_stamp = marker_array.markers[0].header.stamp;` be any different from `const RosTime base_stamp;`?

Reply:
> my understanding of ROS messages in C++ is that all members are default-constructed

That's remains true.

However, in ROS 2, `Time` is measured w.r.t. a given clock. That clock is assumed to be the ROS clock when a `Time` instance is implicitly constructed from a `builtin_interfaces/msg/Time` message (see [here](https://github.com/ros2/rclcpp/blob/06465ba827da0ce018199fb0f80d15ffe6fc26fd/rclcpp/include/rclcpp/time.hpp#L60-L69)), but it is assumed to be the system clock when a `Time` instance is "default constructed" (see [here](https://github.com/ros2/rclcpp/blob/06465ba827da0ce018199fb0f80d15ffe6fc26fd/rclcpp/include/rclcpp/time.hpp#L54)). `Time` instances measured against different clocks cannot be compared. Thus, by instantiating `base_stamp` from a default constructed `builtin_interfaces/msg/Time` message, it will be correctly bound to the ROS clock and comparison will be possible.

AFAIK there's no other portable way to express this. But we could drop portability and use conditional compilation like elsewhere too.

Reply:
I would rather a conditional compilation approach like
```
#if ROS2
const RosTime base_time(0, RCL_ROS_TIME);
#else ROS1
const RosTime base_time;
#endif
```
to make the clock type explicit rather than hiding it in type conversion

Reply:
See https://github.com/calderpg/common_robotics_utilities/pull/9/commits/fcea49e01d4f75a0b0b74fad7a6c1f9e8c0cf7e7.